validphys2/src/validphys/config.py

```diff
@@ -1602,6 +1603,8 @@ def produce_processed_data_grouping(
         if data_grouping is None:
             # fallback to old default behaviour, but still record to lockfile
             data_grouping = self.parse_data_grouping("standard_report")
+            if use_thcovmat_in_fitting or use_thcovmat_in_sampling:
+                data_grouping = self.parse_data_grouping("thcovmat_fit")
```

Comment:
Add this new behavior to the docstring.

And, since you removed `by fitting_group` add a comment somewhere to say that the default fitting group is equal to the the `standard_report` (because a fit is a standard report after all, but better to be explicit since it is not very intuitive).